chips/msp432/src/chip.rs

```diff
@@ -0,0 +1,83 @@
+use core::fmt::Write;
+use cortexm4;
+use kernel::Chip;
+
+use crate::nvic;
+use crate::uart;
+
+pub struct Msp432 {
+    mpu: cortexm4::mpu::MPU,
+    userspace_kernel_boundary: cortexm4::syscall::SysCall,
+    systick: cortexm4::systick::SysTick,
+}
+
+impl Msp432 {
+    pub unsafe fn new() -> Msp432 {
+        Msp432 {
+            mpu: cortexm4::mpu::MPU::new(),
+            userspace_kernel_boundary: cortexm4::syscall::SysCall::new(),
+            systick: cortexm4::systick::SysTick::new_with_calibration(48_000_000),
+        }
+    }
+}
+
+impl Chip for Msp432 {
+    type MPU = cortexm4::mpu::MPU;
+    type UserspaceKernelBoundary = cortexm4::syscall::SysCall;
+    type SysTick = cortexm4::systick::SysTick;
+
+    fn service_pending_interrupts(&self) {
+        unsafe {
+            loop {
+                if let Some(interrupt) = cortexm4::nvic::next_pending() {
+                    match interrupt {
+                        nvic::USCI_A0 => uart::UART0.handle_interrupt(),
+                        _ => {
+                            panic!("unhandled interrupt {}", interrupt);
+                        }
+                    }
+
+                    let n = cortexm4::nvic::Nvic::new(interrupt);
+                    n.clear_pending();
+                    n.enable();
+                } else {
+                    break;
+                }
+            }
+        }
+    }
+
+    fn has_pending_interrupts(&self) -> bool {
+        unsafe { cortexm4::nvic::has_pending() }
+    }
+
+    fn mpu(&self) -> &cortexm4::mpu::MPU {
+        &self.mpu
+    }
+
+    fn systick(&self) -> &cortexm4::systick::SysTick {
+        &self.systick
+    }
+
+    fn userspace_kernel_boundary(&self) -> &cortexm4::syscall::SysCall {
+        &self.userspace_kernel_boundary
+    }
+
+    fn sleep(&self) {
+        unsafe {
+            cortexm4::scb::unset_sleepdeep();
```

Comment:
Why do you need this?

Reply:
Ohh, another copy-paste bug from a different platform^^
I'll remove this line.